blog/2023-04-23-json.md

```diff
@@ -0,0 +1,172 @@
+---
+title: "Json Takes Flight"
+description: Why Wing has a built-in Json primitive
+authors: 
+  - hasanaburayyan
+tags: [cloud-oriented programming, winglang, resources, preflight, inflight, iac, security]
+hide_table_of_contents: true
+---
+
+Picture this: you're in the zone, crafting some spectacular code like the rock star cloud developer you are. You know that to truly harness the power of the cloud ecosystem, your system needs to be distributed. After an all-night coding spree, you've created two nifty services lovingly named service1 and service2. Now, service1 must produce a message and send it to service2. So, it's time to choose a message format standard. Given that it's 2023, XML is off the table, making JSON the clear winner.
+
+At this point, something odd happens in your development journey. You pause to figure out how to use JSON in your chosen programming language, only to discover that you need to import a library. Sure, it's manageable, but now you have to consider the implications of using this JSON library and compare it to others. Of course, you'll make the right choice, you are a rock star, after all. However, this experience leaves you wondering: with JSON being so common, shouldn't it be built into any language designed for distributed computing?
+
+## Communication Is Key
+
+When designing a good distributed system, several key aspects come to mind, such as fault tolerance, scalability, security, high availability, and interoperability. In this section, we'll focus on interoperability—the ability of a system to work seamlessly with other systems. This means that communication between systems must adhere to not only standardized protocols but also data formats.
+
+Wing has been tackling the complexities of distributed computing in the cloud from various angles. One of the most notable angles is the concept of inflight functions, which serve as Wing's distributed computing primitives. These are code blocks packaged up and executed on a compute platform, playing a crucial role in Wing's vision for cloud-oriented programming. As such, it's vital to enable smooth communication between these functions, and what better way to do that than with JSON!
+
+JSON has become the de facto standard for data interchange on the web. As a minimal, text-based format that is a subset of JavaScript, JSON is easy to read and focuses on solving a single problem: data interchange. Its simplicity and widespread adoption make JSON a clear choice for incorporating into Wing, helping us achieve the vision of a truly cloud-oriented programming language.
+
+## Introducing Wing's Json
+
+Wing has decided to treat JSON as a first-class citizen with built-in primitives. Let's take a look at a simple example:
+
+```js
+let j = {
+  wow: "so easy"
+};
+
+// Optionally we can also add the Json keyword prefix
```

Comment:
I saw @eladb commented on Vercel that this is redundant. I actually think it is good to keep, because later in the article it makes sense when you explain mutability and add the `MutJson` keyword.

Reply:
Ok fair enough

Reply:
@Chriscbr I love the insight, I did not really think about it when I was updating examples. Let me swap the direction these are in